Concepts.md

```diff
@@ -0,0 +1,140 @@
+# Crosscutting Concepts
+
+![Crosscutting Concepts](./diagrams/08-crosscutting-concepts.drawio.svg)
+
+## Domain concepts
+
+- [WikibaseDataModel](https://github.com/wmde/WikibaseDataModel)
+- [WikibaseDataModelJavaScript](https://github.com/wmde/WikibaseDataModelJavaScript)
+- [WikibaseDataModelTypes](https://github.com/wmde/WikibaseDataModelTypes)
+
+You can read more about Wikibase's data model in the [Glossary](./Glossary.md#entity).
+
+## User Experience concepts (UX)
+
+### User Interface
+
+User interfaces which were build prior to 2017 use either [OOUI](https://www.mediawiki.org/wiki/OOUI) or [jQuery UI](https://jqueryui.com).  
+[OOUI](https://www.mediawiki.org/wiki/OOUI) is a MediaWiki core library (developed and supported by the WMF) that contains a set of ready-to-use widgets, layouts, windows and a server-side component that generates compatible output in PHP for cases where JavaScript is not supported.
+
+There are several applications which are written in [Vue](https://vuejs.org/) and [TypeScript](https://www.typescriptlang.org/).  
+JavaScript and CSS dependencies are managed through npm. Optimization and minification of files is done through [Webpack](https://v4.webpack.js.org/).  
+There is a central Vue components library in development as part of the [Wikidata/Wikibase Design System](https://wmde.github.io/wikit/). In the future, ideally, all frontend components will be using it.
+
+The project Termbox, which delivers the terms section of an item page on mobile, uses Vue and [Nuxt](https://nuxtjs.org/) to achieve [server side rendering](https://wikitech.wikimedia.org/wiki/WMDE/Wikidata/SSR_Service).
```

Comment:
Termbox doesn't use Nuxt. We looked into it briefly when Termbox development started, but while it worked great for server-side rendering whole pages, SSR for partials such as Termbox was oddly difficult.

```suggestion
The project Termbox, which delivers the terms section of an item page on mobile, uses Vue and [vue-server-renderer](https://www.npmjs.com/package/vue-server-renderer) to achieve [server side rendering](https://wikitech.wikimedia.org/wiki/WMDE/Wikidata/SSR_Service).
```

Reply:
:scream: I've lived a lie